core-backend/common/src/memory.rs

```diff
@@ -47,22 +50,52 @@ pub enum ProcessAccessError {
 
 #[derive(Debug, Clone, derive_more::From)]
 pub enum MemoryAccessError {
-    #[from]
     Memory(MemoryError),
-    #[from]
+    ProcessAccess(ProcessAccessError),
     RuntimeBuffer(RuntimeBufferSizeError),
     // TODO: remove #2164
     Decode,
-    GasLimitExceeded,
-    GasAllowanceExceeded,
 }
 
-impl From<ProcessAccessError> for MemoryAccessError {
-    fn from(err: ProcessAccessError) -> Self {
-        match err {
-            ProcessAccessError::OutOfBounds => MemoryError::AccessOutOfBounds.into(),
-            ProcessAccessError::GasLimitExceeded => Self::GasLimitExceeded,
-            ProcessAccessError::GasAllowanceExceeded => Self::GasAllowanceExceeded,
+impl BackendExternalitiesError for MemoryAccessError {
```

Comment:
If it's implemented for MemoryAccessError then `Externalities` in name is not correct